Stage 1

jduncan017 opened the pull request:
Since my app has more than 2 pages please review the home page and the contact page ("/" & "/contact"). Contact has the required api which is under src/app/api.

FiLLa-FaS commented:
Hi! Thanks for sending your project for review, it looks great! You've got more than 80 points, so I'm glad to accept your project already, but you still can fix the issues below to improve your project a little more. Good luck!🙌🚀

# Summary:
### Performance Criteria

#### Markup and JSX
- [x] <!--7.4--> The project is adapted for different screen resolutions and looks correctly on all intermediate resolutions. There is no horizontal scrolling on resolutions of `320px` and up. The scrollbar can't be hidden using the `overflow: hidden` declaration.
- [x] <!--7.26--> Navigation between pages and links to external resources are working correctly: there are no broken links or links leading to an empty anchor on the page and external links open in a new tab. All project pages can be accessed, and hidden blocks can be displayed.
- [x] <!--7.26--> Buttons, forms, and modal windows from the UI toolkit are used.
- [ ] <!--3.63--> Classes are named according to BEM specifications.
    * Please don't use dots and capital letters in class names throughout the project
    * BEM elements shouldn't be used outside their blocks (for example `global__page-borders` shouldn't be used outside the `global` block. Please fix all similar cases **throughout the project**
    * `banner__textbox_text` should be a BEM element, not a modifier
- [ ] <!--3.63--> Semantic HTML is used, meaning that semantic tags are used. All elements are used correctly (e.g. a paragraph must be a paragraph, a list must be a list). The DOM tree structure doesn't consist only of `<div />` containers.
    * You need to use more semantic tags like `header`, `footer`, `main`, `section` and `nav` tags in your project instead of `div` tags
- [x] <!--3.63--> The correct approach is chosen for positioning elements, and is described using the correct syntax. For example, when positioning an element absolutely, its parent block is relatively positioned.
- [x] <!--3.63--> `flex` or `grid`  layout are used to arrange elements.
- [x] <!--3.63--> Infrastructural project files are created using CRA.
- The markup has been converted into JSX:
  - [x] <!--3.63--> The markup is included within parentheses `( )`.
  - [x] <!--3.63--> The markup has been moved into the corresponding components.
- [x] <!--3.63--> Your project contains:
  - [x] An `images`
  - [x] A `components` folder with JS and CSS files for the components.
  - [ ] A `fonts` folder.
- [x] <!--3.63--> There are no linting errors or warnings.

#### React
- React components:
  - [x] <!--3.63--> Hooks are not used inside conditional statements or loops.
  - [x] <!--3.63--> Hooks are called in a component's main function.
  - [x] <!--3.63--> For class components, effects are described inside the component lifecycle methods.
- [ ] <!--7.26--> Asynchronous API requests:
  - Requests can be made through the Fetch API or by using XMLHttpRequest. Third-party libraries (such as axios or jQuery) are not used.
  - API requests are contained in a separate file.
  - The chain for processing promises ends with a `catch()` block.
  - The first `then()` handler returns `res.json`.
      * Api requests should be moved from ContactForm component to separate file
- [x] <!--3.63--> Naming:
  - camelCase is used for function and variable names.
  - Only nouns are used as variable names.
  - Variable names clearly describe what is stored in them. If the project has several variables with similar data, then those variables have unique but descriptive names.
  - Descriptive names are used for functions, which reflect what they do.
  - Function names start with a verb.
  - JS classes and functional components are named using nouns and start with a capital letter.
  - Names must not include inappropriate or unclear abbreviations.
  - Сustom hook names start with `use`
- [x] <!--3.63--> No third-party JavaScript libraries are used unless absolutely necessary. If third-party libraries are connected, then they are used appropriately.

### Best Practices

#### Markup and JSX
- [x] <!--1.25--> The design reuses components wherever possible.
- [x] <!--1.25--> Fonts are connected using `@font-face`.
- [ ] <!--1.25--> The icons are exported in SVG format.
    * All icons (including logo) should be in svg format
- [x] <!--1.25--> Form elements should be highlighted when focused.
- [x] <!--1.25--> The form must have placeholders,  and there are required properties for the fields.
- [x] <!--1.25--> The use of `reset.css` is prohibited.

#### React
- [x] <!--1.25--> The initial state of state variables contain the correct data type.
- [x] <!--1.25--> API requests are described inside the `App` component.
- [x] <!--1.25--> There is no memory leak when hanging handlers. All handlers added with addEventListener are removed when the component is unmounted.
- [x] <!--1.25--> API error handling:
  - [x] The user receives a message in case of an error.
- [ ] <!--1.25--> Non-variable values (hard-coded constants) are named in all capital letters and stored in a separate configuration file.
    * All constants should be named with all capitals letters
- [x] <!--1.25--> If you are using popups, then correct the opening and closing functionality is implemented. Popups can be closed by clicking on the cross button, overlay, or by pressing the ESC key.

### Recommendations

#### Markup and JSX
- [x] <!--0.83--> System fonts are connected as alternatives to each of your fonts
- [x] <!--0.83--> Images have an `alt` attribute containing appropriate values.
- [x] <!--0.83--> Raster and vector images have been optimized.

#### React
- [x] <!--0.83--> For internal links in the application, components from the react-router library are used.
- [x] <!--0.83--> Semantically correct blocks are used for components. No `<div />` or other unnecessary HTML tags are used for components that consist of single-level blocks
- [x] <!--0.85--> The code is clean and easy to understand:
  - The code is readable and structured clearly. Some parts of the code are explained with comments if needed.
  - There is no extra code: for example, a variable is declared but not used, or there is some kind of redundant logic.
  - The code is formatted in the same way, the indentation hierarchy is respected.

## Number of points:  83

jduncan017 commented:
Thank you for the feedback! I did my best to address the issues you've noted:
1. I updated HTML semantics and BEM classNames. 
2. I left the global classnames; I like that they save on code while allowing for reusable & easy to maintain effects like page borders and box shadows.
3. In Next.js fonts is just a single file that is used throughout the code so it doesn't need a folder, but I did move it to the app folder.
4. Api request files were moved to their matching api routes
5. I changed a few png files to svg where it made sense based on image complexity, but not all logos worked as svg.
6. Updated constants to all caps
